src/rules/no-anonymous-default-export.js

```diff
@@ -0,0 +1,84 @@
+/**
+ * @fileoverview Rule to disallow anonymous default exports.
+ * @author Duncan Beevers
+ */
+
+const defs = {
+  ArrayExpression: {
+    option: 'allowArray',
+    message: 'an array',
+  },
+  ArrowFunctionExpression: {
+    option: 'allowArrowFunction',
+    message: 'an arrow function',
+  },
+  ClassDeclaration: {
+    option: 'allowAnonymousClass',
+    message: 'an anonymous class',
+    forbid: (node) => !node.declaration.id,
+  },
+  FunctionDeclaration: {
+    option: 'allowAnonymousFunction',
+    message: 'an anonymous function',
+    forbid: (node) => !node.declaration.id,
+  },
+  Literal: {
+    option: 'allowLiteral',
+    message: 'a literal',
+  },
+  ObjectExpression: {
+    option: 'allowObject',
+    message: 'an object expression',
+  },
+}
+
+const defaultOptions = Object.keys(defs).
+  map((key) => defs[key]).
+  reduce((acc, def) => {
+    acc[def.options] = false
```

Comment:
This `defaultOptions` object doesn't strictly need to be constructed. I created it (poorly) for parity with the documentation. Since it doesn't affect behavior, I'm rewinding my decision here and removing this.